service/src/main/java/bio/terra/workspace/service/iam/SamService.java

```diff
@@ -1051,14 +1052,19 @@ public AuthenticatedUserRequest getOrCreatePetSaCredentials(
    * {@code getOrCreatePetSaEmail}, this will not create the underlying service account. It may
    * return the email of a service account which does not exist.
    */
-  public ServiceAccountName constructUserPetSaEmail(
+  public Optional<ServiceAccountName> constructUserPetSaEmail(
       String projectId, String userEmail, AuthenticatedUserRequest userRequest)
       throws InterruptedException {
     UsersApi usersApi = samUsersApi(userRequest.getRequiredToken());
     try {
-      String subjectId = SamRetry.retry(() -> usersApi.getUserIds(userEmail).getUserSubjectId());
+      UserIdInfo userId = SamRetry.retry(() -> usersApi.getUserIds(userEmail));
+      if (userId == null) {
```

Comment:
// If userId is null, userEmail is a group, not a user. (getUserIds returns 204 with no response body, which translates to userID = null.)